linkatos/activities.py

```diff
@@ -33,29 +37,43 @@ def event_consumer(url_cache_list, slack_client, bot_id,
         if is_unfurled(event):
             return url_cache_list
 
-        if event['type'] == 'message' and not 'username' in event:
+        if event['type'] == 'message' and 'username' not in event:
             new_url_cache = parser.parse_url_message(event)
 
             if is_url(new_url_cache) and is_not_from_bot(bot_id,
                                                          new_url_cache['user']):
                 url_cache_list.append(new_url_cache)
                 printer.ask_confirmation(new_url_cache, slack_client)
+                return url_cache_list
 
             if message.to_bot(event['text'], bot_id):
                 list_request = parser.parse_list_request(event)
+                purge_request = parser.parse_purge_request(event)
 
-                if 'type' in list_request and list_request['type'] == 'list_request':
+                if list_request is not None and list_request['type'] == 'list_request':
                     printer.list_cached_urls(url_cache_list,
                                              list_request['channel'],
                                              slack_client)
+                    return url_cache_list
+
+                if purge_request is not None and \
```

Comment:
I agree. I've been thinking about this but will do it after documenting the functions.
Do you think this is necessary for this PR?